lib/fs.js

```diff
@@ -572,17 +571,52 @@ function openSync(path, flags, mode) {
  * @param {{
  *   type?: string;
  *   }} [options]
- * @returns {Promise<Blob>}
+ * @param {(
+ *   err?: Error,
+ *   blob?: Blob
+ *   ) => any} callback
+ * @returns {void}
+ */
+function openAsBlob(path, options, callback) {
+  if (callback === undefined) {
+    callback = options;
+    options = kEmptyObject;
+  } else {
+    options ??= kEmptyObject;
+  }
+
+  validateObject(options, 'options');
+  const type = options.type || '';
+  validateString(type, 'options.type');
+  validateFunction(callback, 'cb');
+
+  let blob;
+  let cbError = null;
+  try {
+    blob = createBlobFromFilePath(pathModule.toNamespacedPath(path), { type });
+  } catch (err) {
+    // Permission errors should be thrown instead of being passed to callback
+    if (err.code === 'ERR_ACCESS_DENIED') {
+      throw err;
+    }
+    cbError = err;
+  }
+  callback(cbError, blob);
```

Comment:
If this is to mimmic the promise API and pretend to be async this needs a process.nextTick or queueMicrotask